lib/buffer.js

```diff
@@ -177,10 +173,12 @@ Buffer.from = function(value, encodingOrOffset, length) {
     return fromArrayBuffer(value, encodingOrOffset, length);
 
   if (value == null)
-    throw new TypeError(kFromErrorMsg);
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'first argument',
+      ['string', 'buffer', 'arrayBuffer', 'array', 'array-like object'], value);
 
   if (typeof value === 'number')
-    throw new TypeError('"value" argument must not be a number');
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'value', 'not number',
```

Comment:
What does the string end up being here ?  I think this may be the first case were we have the 'not' cases versus listing out what it should be.